packages/yarnpkg-core/sources/Workspace.ts

```diff
@@ -157,6 +157,29 @@ export class Workspace {
     return workspaceList;
   }
 
+  /**
+   * Find workspaces which include the current workspace as a dependency/devDependency recursively.
+   *
+   * @returns all the workspaces marked as dependents
+   */
+  async getRecursiveWorkspaceDependents() {
+    await this.project.restoreInstallState();
```

Comment:
This shouldn't be in this function (the restore call should be made at the beginning of the commands that require it, otherwise it may have a hefty perf impact and cause weird behaviors).
```suggestion
  getRecursiveWorkspaceDependents() {
```

Reply:
Fixed in 711507b73bf4475858bdcd05d4d4aa4bb6bd5335.